README.md

```diff
@@ -1 +1,1118 @@
+
+Rekor has a number of properties that make it useful as an attestation store; first, its underlying storage is a robust, append-only log – new entries can be added, but old entries cannot be updated or deleted.
+Rekor’s implementation also provides the ability to demonstrate that an entry is actually included in the log, which users can verify through what’s called an inclusion proof.
+Similarly, by recording certain metadata about the log, users can also verify over time that the log hasn’t been tampered with and that only new entries have been created (through a consistency proof).
+
+Rekor is used to store container image signatures, along with the public key or certificate that's needed to verify the signature. It can also store [in-toto attestations](https://in-toto.io/), along with a few other [built-in record types](https://docs.sigstore.dev/rekor/sign-upload/).
+
+#### Sigstore: Fulcio Certificate Authority
+
+Fulcio is a certificate authority for code-signing -- it issues short-lived (10 minute) certificates that anyone can use to sign images or attestations.
+The [official documentation](https://docs.sigstore.dev/fulcio/certificate-issuing-overview/) goes into more detail, but at a high level this is the flow for acquiring a code-signing certificate:
+
+1. Acquire an id token from an issuer that Fulcio is configured to trust. This demo uses the GitHub Actions id token
+1. Generate a temporary public/private key pair - `cosign` will do this automatically
+1. Prove that you control the key pair by doing one of the following:
+   - Create a certificate signing request (CSR)
+   - Grab the `sub` claim from the id token and sign it
+1. Make a request to Fulcio with the id token and the proof from the previous step
+1. Fulcio will validate the id token and check the provided proof
+1. If the request is valid, Fulcio will issue a code-signing certificate and populate the certificate fields with information from the id token
+   - When using a GitHub Actions OIDC token, the issuer will be `https://token.actions.githubusercontent.com` and the subject will be the value of the `job_workflow_ref` claim
+1. Depending on the deployment type, Fulcio will also submit a request to a certificate transparency log to include the certificate in the logs. The response from the certificate transparency log will include a signed timestamp that will be added to the certificate.
+
+After receiving the certificate, `cosign` or other tools can sign images or attestations with the private key generated earlier, and then upload the signature and certificate to Rekor.
+Once the signing is finished, the key pair can be discarded, as the certificate will be in Rekor for validation later.
+
+For a more concrete example, this is a sequence diagram that shows how a GitHub Actions workflow might build & sign a container image:
+```mermaid
```

Comment:
This diagram is really useful, but I think GitHub's rendering of larger mermaid documents is still too lacking. It's not easy to navigate and zoom on this when trying to understand the contents. Could we do something like generate a svg from the mermaid diagram, and include that asset in the readme?

Reply:
I converted to svg with the Mermaid CLI. The inline view is about the same, but it's easier to open in a new tab and see the full diagram